parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -229,16 +229,25 @@ fi
 VALIDATION_ERR=0
 
 {{- if IsHostedMaster }}
-RES=$(retrycmd_if_failure 20 1 3 nslookup ${API_SERVER_NAME})
+API_SERVER_DNS_RETRIES=20
+if [[ $API_SERVER_NAME == *.privatelink.* ]]; then
+  API_SERVER_DNS_RETRIES=100
+fi
+RES=$(retrycmd_if_failure ${API_SERVER_DNS_RETRIES} 1 3 nslookup ${API_SERVER_NAME})
 STS=$?
 if [[ $STS != 0 ]]; then
     if [[ $RES == *"168.63.129.16"*  ]]; then
         VALIDATION_ERR=$ERR_K8S_API_SERVER_AZURE_DNS_LOOKUP_FAIL
     else
         VALIDATION_ERR=$ERR_K8S_API_SERVER_DNS_LOOKUP_FAIL
     fi
+    exit $VALIDATION_ERR
+fi
+API_SERVER_CONN_RETRIES=50
+if [[ $API_SERVER_NAME == *.privatelink.* ]]; then
+  API_SERVER_CONN_RETRIES=200
```

Comment:
question: so private cluster node max timeout is (100+200) * (1+3) = 1200 second that is 20min, do we plan to have the same timeout in ccp readiness check in RP?

Reply:
I think we already have that (longer than 20m by default)